### PR TITLE
Add Send/Sync annotation to allow ExternalVector to be send to other threads

### PR DESCRIPTION
**Pull request by VeaaC**
Signed-off-by: Christian Vetter <christian.vetter@here.com>

### CODE REVIEW

File: flatdata-rs/lib/src/storage.rs [hunk 1]
```diff
@@ -13,8 +13,6 @@ use std::{
     sync::Arc,
 };
 
-use diff;
```
Comment (boxdot) [inline flatdata-rs/lib/src/storage.rs:16]: Is diff still used? If not, we could remove it from toml?

File: flatdata-rs/lib/src/storage.rs [hunk 2]
```diff
@@ -195,7 +193,7 @@ pub fn check_resource<T>(
 /// storage.
 #[doc(hidden)]
 pub fn create_external_vector<'a, T>(
-    storage: &'a dyn ResourceStorage,
+    storage: &'a (dyn ResourceStorage + std::marker::Sync + std::marker::Send),
```
Comment (boxdot) [inline flatdata-rs/lib/src/storage.rs:196]: I think Sync and Send are in prelude, so technically no fqn is needed.

**Review by boxdot — APPROVED** [review]

File: flatdata-rs/lib/src/storage.rs (hunk 1 above)
Comment (VeaaC) [reply flatdata-rs/lib/src/storage.rs:16]: Still used, but directly (`dif::xyz`)

File: flatdata-rs/lib/src/storage.rs (hunk 2 above)
Comment (VeaaC) [reply flatdata-rs/lib/src/storage.rs]: Done

**Pull request merged**